src/server/acl/validator.cc

```diff
@@ -86,6 +97,45 @@ namespace dfly::acl {
   return {keys_allowed, AclLog::Reason::KEY};
 }
 
+[[nodiscard]] std::pair<bool, AclLog::Reason> IsUserAllowedToInvokePubSubCommand(
+    bool literal_match, const std::vector<uint64_t>& acl_commands, const AclPubSub& pub_sub,
+    CmdArgList tail_args, const CommandId& id) {
+  const size_t index = id.GetFamily();
+  const uint64_t command_mask = id.GetBitIndex();
+  DCHECK_LT(index, acl_commands.size());
+
+  const bool command = (acl_commands[index] & command_mask) != 0;
+
+  if (!command) {
+    return {false, AclLog::Reason::COMMAND};
+  }
+
+  auto match = [](const auto& pattern, const auto& target) {
```

Comment:
it's hard to read when everything is auto. I assume that at least some of these arguments are string_view, which is a simple type to add. 
Can you please reduce auto usage with lambdas?